source/text/Format.cpp

```diff
@@ -377,6 +377,64 @@ string Format::PlayTime(double timeVal)
 
 
 
+// Convert an ammo count into a short string for use in the ammo display.
+// Only the absolute value of a negative number is considered.
+string Format::AmmoCount(int64_t value)
+{
+	if(fabs(value) > SCIENTIFIC_THRESHOLD)
+	{
+		ostringstream out;
+		out.precision(2);
+		out << static_cast<double>(value);
+		return out.str();
+	}
+
+	int64_t absolute = abs(value);
+
+	if(absolute < 10000)
+		return to_string(value);
+
+	string result;
+	result.reserve(5);
+
+	// Handle numbers bigger than a million.
+	static const vector<char> SUFFIX = {'T', 'B', 'M', 'k'};
+	static const vector<int64_t> THRESHOLD = {1000000000000ll, 1000000000ll, 1000000ll, 1000ll};
```

Comment:
```suggestion
	static const vector<int64_t> THRESHOLD = {1000'000'000'000ll, 1000'000'000ll, 1000'000ll, 1000ll};
```

Reply:
A decent chunk of this code is very similar to `Format::Credits` and I don't want the style to diverge.
We also don't appear to do this anywhere else.

Reply:
We might be approaching the point where it would make sense to decouple the common logic to a helper function.

Reply:
It is possible to consolidate some of the code, and while it may reduce the number of places where style changes like this need to be applied, I think the overall result of consolidation would be substantially less readable than the current version, so I don't think that would be a good change.

Reply:
I think it would make sense to move the thresholds and suffixes to the namespace, though, as they are shared. I'm not a huge fan of local static variables, either.

Reply:
The other method doesn't have a thousand threshold, only M, B, and T.
We could still move the vectors to the namespace and this method could make copies with the extra elements, but I'm not sure that's really any better than this, it's arguably worse because then some of the values this method expects to see are defined locally but others are in the namespace.